src/lib.rs

```diff
@@ -620,17 +676,24 @@ pub fn init(options: &InitOptions) -> Result<(ZWaveManager, mpsc::Receiver<ZWave
     let (mut zwave_manager, rx) = ZWaveManager::new(manager);
     try!(zwave_manager.add_watcher());
 
+    let mut fs_watcher = try!(FsWatcher::new());
+
     let devices = options.devices.clone().unwrap_or(get_default_devices());
-    for device in devices.iter() {
+
+    for device in devices {
         try !(
             fs::File::open(&device).map_err(|err| Error::CannotReadDevice(device.clone(), err))
         );
-        //println!("found device {}", device);
 
+        debug!("[OpenzwaveStateful] found device {}", device);
+
+        try!(fs_watcher.watch(&device));
```

Comment:
I think that there are a variety of reasons why this could fail, and we should probably just report and ignore the failure.

1 - The kernel could be configured with inotify turned off.
2 - The max allocated inotify resources could already be in use (the process which use using openzwave may have otherwise exhausted the resources)